faction.cpp

```diff
@@ -886,7 +1110,11 @@ void Faction::DiscoverItem(int item, int force, int full)
 		}
 	}
 	if (force) {
-		itemshows.Add(ItemDescription(item, full));
+		AString *desc = ItemDescription(item, full);
```

Comment:
This was leaking memory before because ItemDescription() returns a new'd AString pointer.